apps/web/lib/signup/getServerSideProps.tsx

```diff
@@ -63,15 +63,18 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
   // no token given, treat as a normal signup without verification token
   if (!token) {
     return {
-      props: JSON.parse(
-        JSON.stringify({
-          ...props,
-          prepopulateFormValues: {
-            username: preFillusername || null,
-            email: prefilEmail || null,
-          },
-        })
-      ),
+      props: {
+        trpcState: ssr.dehydrate(),
+        ...JSON.parse(
+          JSON.stringify({
+            ...props,
+            prepopulateFormValues: {
+              username: preFillusername || null,
+              email: prefilEmail || null,
+            },
+          })
+        ),
+      },
```

Comment:
We have a lot of key/value pairs that need to be fixed for the social media preview, not just one. Need to figure out how to server-side render the translations for the meta tags. Is this the right way to da that?

Reply:
Yes `ssrInit()` will call [serverSideTranslations](https://github.com/snehalbaghel/cal.com/blob/add59896f57d638058fedf7e395977a574f718db/apps/web/server/lib/ssr.ts#L60) and set it to queryClient cache so it doesn't have to refetch on the client. Other pages already do it this way, I noticed it's not being called for these two pages. I'll do an exhaustive check and come back.

Let me know if the approach is fine? Do we need to do it some other way?

Reply:
@anikdhabal This [issue](https://github.com/i18next/next-i18next/issues/1532) might be relevant.